source/postprocess/visualization.cc

```diff
@@ -572,12 +656,18 @@ namespace aspect
       base_variables.initialize_simulator (this->get_simulator());
 
       std::unique_ptr<internal::MeshDeformationPostprocessor<dim> > mesh_deformation_variables;
+      internal::BaseAdjointVariablePostprocessor<dim> adjoint_base_variables;
+      adjoint_base_variables.initialize_simulator (this->get_simulator());
 
       DataOut<dim> data_out;
       data_out.attach_dof_handler (this->get_dof_handler());
       data_out.add_data_vector (this->get_solution(),
                                 base_variables);
 
+      if (this->get_adjoint_problem() == true)
+        data_out.add_data_vector (this->get_current_adjoint_solution(),
+                                  adjoint_base_variables);
+
```

Comment:
Couldnt you always add this if the nonlinear solver scheme is `stokes_adjoint`? After every timestep (when this function is called) there should be a normal solution, and an adjoint solution to output, so lets just output everything in a single operation. Or am I missing something and not all information is still available at the end of the timestep (e.g. is something overwritten in the adjoint phase)?

Reply:
Yes and no - I think this is a good suggestion and allows to further remove instances of where I call get_adjoint_problem. I think one thing I need to change for this though is that after the forward run I do postprocess() so that the dynamic topography is calculated. I don't actually need the other postprocessors at this point and it would be better to postprocess everything else after the adjoint equations are solved as well because then there's actually something here to plot. Can I run just the DT postprocessor ... or how do I make sure that that's accessible after the forward solution is done?